lib/flow_commerce/flow_api_v0_client.rb

```diff
@@ -35,8 +35,11 @@ module Constants
 
         def initialize(url, opts={})
           @url = HttpClient::Preconditions.assert_class('url', url, String)
+          @base_url = URI(url)
           @authorization = HttpClient::Preconditions.assert_class_or_nil('authorization', opts.delete(:authorization), HttpClient::Authorization)
           @default_headers = HttpClient::Preconditions.assert_class('default_headers', opts.delete(:default_headers) || {}, Hash)
+          @http_handler = opts.delete(:http_handler) || HttpClient::DefaultHttpHandler.new
```

Comment:
the reason I chose not to is that I don't know what the class will be (you can provide your own custom class). What we would want to assert is that the class implements the http handler interface - but there's no native way to do this in ruby (other than explicitly checking if methods are defined) - thus chose to leave w/out an assertion